packages/typespec-ts/test/unit/modelsGenerator.spec.ts

```diff
@@ -346,6 +346,166 @@ describe("Input/output model type", () => {
     });
   });
 
+  describe("decimal generation", () => {
+    it("should handle decimal -> number", async () => {
```

Comment:
Map the decimal to `number` in JS directly, and will 
- add below comments 
```
Please note the field was supposed to be a decimal but JavaScript does not have a native 'BigDecimal' data type.
So it was converted to a number instead. It is recommended to use a third-party library like 'decimal.js' to handle 
any calculations.
```
- report a warning 
```
Please note the decimal type will be converted to number. If you strongly care about precision you can use @encode to encode it as a string for the property - {your prop name}.
```